crates/http-api-bindings/src/embedding/mod.rs

```diff
@@ -1,18 +1,31 @@
 mod llama;
 mod openai;
+mod rate_limit;
 mod voyage;
 
 use core::panic;
 use std::sync::Arc;
 
 use llama::LlamaCppEngine;
+use rate_limit::RateLimitedEmbedding;
 use tabby_common::config::HttpModelConfig;
 use tabby_inference::Embedding;
+use tracing::debug;
 
 use self::{openai::OpenAIEmbeddingEngine, voyage::VoyageEmbeddingEngine};
 
 pub async fn create(config: &HttpModelConfig) -> Arc<dyn Embedding> {
-    match config.kind.as_str() {
+    let rpm = if let Some(limit) = &config.request_limit {
+        limit.request_per_minute
+    } else {
+        debug!(
+            "No request limit specified for model {}, defaulting to 600 rpm",
+            config.kind
+        );
+        600
```

Comment:
make this a default option in `config.rs`:

example: https://demo.tabbyml.com/files/github/TabbyML/tabby/-/blob/f2cf549ad9ef01338ce30626095108d0ca1bf5ed/crates/tabby-common/src/config.rs#L243